addons/website/static/src/core/interaction.js

```diff
@@ -124,7 +131,7 @@ export class Interaction {
     waitFor(fn) {
         return new Promise(async (resolve) => {
             const result = await fn.call(this);
-            if (!this.isDestroyed) {
+            if (this.isReady && !this.isDestroyed) {
                 resolve(result);
```

Comment:
Shouldn't the resolve be done if not ready ? Otherwise we'll end up with calls from setup never completing